### PR TITLE
Fix updating years in license file (#65)

### PR DESCRIPTION
**Pull request by aostrowski**

### CODE REVIEW

File: pre_commit_hooks/insert_license.py [hunk 1]
```diff
@@ -89,10 +89,13 @@ def main(argv=None):
     return 0
 
 
-def _replace_year_in_license_with_current(plain_license: list[str]):
+def _replace_year_in_license_with_current(plain_license: list[str], filepath: str):
     current_year = datetime.now().year
     for i, line in enumerate(plain_license):
-        plain_license[i] = re.sub(r"\b\d{4}\b", str(current_year), line)
+        updated = try_update_year(line, filepath, current_year, introduce_range=False)
```
Comment (Lucas-C) [inline pre_commit_hooks/insert_license.py:95]: If I understand the code correctly, at this point we found an existing license header with some year mentioned,
but not a year range, and we are going to convert the year into a year range?
Why doing so?
And do we have a unit test covering year range insertion ?

File: tests/insert_license_test.py [hunk 2]
```diff
@@ -99,16 +105,23 @@ def test_insert_license(license_file_path,
...
+                ("module_with_license.groovy", "//"),
+                ("module_with_license_and_numbers.py", "#"),
+                ("module_with_year_range_in_license.py", "#"),
+                ("module_with_spaced_year_range_in_license.py", "#"),
+            ),
         ),
-    )),
+    ),
 )
 def test_insert_license_current_year_already_there(license_file_path,
```
Comment (Lucas-C) [inline tests/insert_license_test.py:126]: It's nice to have a dedicated test function for `--use-current-year` usages 😊

However, this function does not validate the content of the file post-insertion,
as `test_insert_license()` or `test_fuzzy_match_license()`.
Could you please add a `new_src_file_expected` parameter to this function?

**Review by Lucas-C — COMMENTED** [review]
The code is great, thank you for the improvements!
I made a couple of comments.

A minor point: there is no docstring for the `update_year_range` parameter in `license_found()` function definition.

File: tests/insert_license_test.py (hunk 2 above)
Comment (aostrowski) [reply tests/insert_license_test.py:126]: If the current year is already there we don't expect any modification. I added a check to verify the contents of the file are identical before and after the `insert_license()` call.

File: pre_commit_hooks/insert_license.py (hunk 1 above)
Comment (aostrowski) [reply pre_commit_hooks/insert_license.py:95]: This code is for verifying the correct year is in the license file. 
The `introduce_range=False` argument states that if the license contains a year it should not be converted to a range. So, if a range is already there, we will update it. But if the license file contains just a single year, we will replace this year with the current one.
This is different from another call to `try_update_year(..., introduce_range=True)` from `try_update_year_range`. We do want to introduce a range there, as this means a source file was modified some past year and we want it to also contain the current year.

Comment (Lucas-C) [reply pre_commit_hooks/insert_license.py:95]: Ok, thank you for the explanation, that's great!

File: tests/insert_license_test.py (hunk 2 above)
Comment (Lucas-C) [reply tests/insert_license_test.py:126]: Perfect

**Pull request merged**